cmd/init.go

```diff
@@ -0,0 +1,278 @@
+	if err := setupDirPathForClientConfig(dirPath, clientConfig); err != nil {
+		return err
+	}
+	marshalledClientConfig, err := yaml.Marshal(clientConfig)
+	if err != nil {
+		return err
+	}
+	filePath := getClientConfigPath(dirPath)
+	filePermission := 0o660
+	return os.WriteFile(filePath, marshalledClientConfig, fs.FileMode(filePermission))
+}
+
+func setupDirPathForClientConfig(dirPath string, clientConfig *config.ClientConfig) error {
+	directoryPermission := 0o750
+	for _, namespace := range clientConfig.Namespaces {
+		namespaceDirPath := path.Join(dirPath, namespace.Name)
+		namespaceDatastoreDirPath := path.Join(namespaceDirPath, "resources")
+		namespaceJobDirPath := path.Join(namespaceDirPath, "jobs")
+		paths := []string{namespaceDirPath, namespaceDatastoreDirPath, namespaceJobDirPath}
+		for _, p := range paths {
+			if err := os.RemoveAll(p); err != nil {
+				return fmt.Errorf("error deleting [%s] for namespace [%s]: %w",
+					p, namespace.Name, err,
+				)
+			}
+			if err := os.Mkdir(p, fs.FileMode(directoryPermission)); err != nil {
+				return fmt.Errorf("error creating [%s] for namespace [%s]: %w",
+					p, namespace.Name, err,
+				)
+			}
+		}
+	}
+	return nil
+}
+
+func askInitClientConfig(l log.Logger, dirPath string) (*config.ClientConfig, error) {
+	output := &config.ClientConfig{}
+	host, err := askInitHost(l)
+	if err != nil {
+		return nil, err
+	}
+	project, err := askInitProject(l)
+	if err != nil {
+		return nil, err
+	}
+	namespaces, err := askInitNamespaces(l, dirPath)
+	if err != nil {
+		return nil, err
+	}
+	output.Version = config.DefaultVersion
+	output.Host = host
+	output.Project = project
+	output.Namespaces = namespaces
+	output.Log = config.LogConfig{
+		Level: config.LogLevelInfo,
+	}
+	return output, nil
+}
+
+func askInitNamespaces(l log.Logger, dirPath string) ([]*config.Namespace, error) {
+	var output []*config.Namespace
+	for {
+		name, err := askInitNamespaceName(l, dirPath)
+		if err != nil {
+			return nil, err
+		}
+		datastoreType, err := askInitNamespaceDatastoreType()
+		if err != nil {
+			return nil, err
+		}
+		namespace := &config.Namespace{
+			Name: name,
+			Datastore: []config.Datastore{
+				{
+					Type:   datastoreType,
+					Path:   path.Join(name, "resources"),
+					Backup: make(map[string]string),
+				},
+			},
+			Job: config.Job{
+				Path: path.Join(name, "jobs"),
+			},
+		}
+		output = append(output, namespace)
+
+		confirmMessage := "Do you want to add another namespace?"
+		confirmHelp := "If yes, then you will be prompted to create another namespace"
+		confirmedToAddMore, err := askToConfirm(confirmMessage, confirmHelp, false)
+		if err != nil {
+			return nil, err
+		}
+		if !confirmedToAddMore {
+			break
+		}
+		l.Info("Adding more namespaces")
+	}
+	return output, nil
+}
+
+func askInitNamespaceDatastoreType() (string, error) {
+	prompt := &survey.Select{
+		Message: "What is the type of data store for this namespace?",
+		Options: []string{
+			"bigquery",
```

Comment:
yes, we support only bigquery. I'm not sure about this myself. but, in the comment from the previous version mentioned these three (can be checked [here](https://github.com/odpf/optimus/blob/b547d74b4c6f285aa47c97b891d42025b9620b74/config/config.go#L29)). so, since I wasn't sure why it was mentioned that way, I just leave it be. if you think it's unnecessary as of now, I will remove the other two.

Reply:
yeah please remove

Reply:
it's addressed